### PR TITLE
Odyssey: Bundle size optimization

### PR DESCRIPTION
**Pull request by kangzj**
## Proposed Changes

The approaches reduces the main bundle size from `~980K` to `~840K`, and overall asset size from `~6.1M` to `~5.1M`.

The PR proposes to

- Externalize `moment.js` and use the WordPress built-in one, which is loaded by default for WP-Admin <- removes `~160KB`

- Remove `calypso\/components\/inline-support-link`
- Remove `calypso\/components\/web-preview` <- this is a major one, which reduces the all asset size by several hundred KBs.
- Remove `calypso\/blocks\/upsell-nudge`
- Remove `calypso\/my-sites\/stats\/mini-carousel`
- Remove `calypso\/blocks\/jetpack-backup-creds-banner`
- Remove `calypso\/components\/data\/query-keyring-connections`
- Remove `calypso\/components\/data\/query-jetpack-modules`
- Remove `calypso\/components\/data\/query-site-keyrings`

- <strike>Async load `calypso/my-sites/stats/promo-cards`</strike>

Related: pejTkB-hT-p2

## Testing Instructions

* Follow instructions in PejTkB-3E-p2
* Ensure Odyssey Stats works well

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by jsnmoon — CHANGES_REQUESTED** [review]
```
section_name                       parsed_size           gzip_size
stats                               -50512 B  (-7.5%)   -14169 B  (-7.3%)

async_section_name                                parsed_size             gzip_size
async-load-calypso-my-sites-stats-promo-cards     +51111 B      (new)   +16355 B      (new)
async-load-calypso-my-sites-stats-wordads         +51024 B  (+202.3%)   +15816 B  (+187.6%)
async-load-calypso-my-sites-stats-summary         +51024 B  (+188.4%)   +15837 B  (+212.3%)
```

Async loading PromoCards for the traffic page unintendedly bloated our `wordads` and `summary` bundles. Could we leave this change out of the PR?

**Comment by kangzj** [comment]
> Async loading PromoCards for the traffic page unintendedly bloated our wordads and summary bundles. Could we leave this change out of the PR? @jsnmoon 

Good catch. Promo Cards gets bundled in these pages instead I think. Reverted in 1087050d69.

File: client/my-sites/stats/site.jsx [hunk 1]
```diff
@@ -178,13 +178,7 @@ class StatsSite extends Component {
 						"Learn more about the activity and behavior of your site's visitors. {{learnMoreLink}}Learn more{{/learnMoreLink}}",
 						{
 							components: {
-								learnMoreLink: (
-									<InlineSupportLink
-										supportContext="stats"
-										showIcon={ false }
-										showSupportModal={ ! isOdysseyStats }
```
Comment (kangzj) [inline client/my-sites/stats/site.jsx:185]: The change is not necessary anymore.

**Review by jsnmoon — APPROVED** [review]
[ICFY](http://iscalypsofastyet.com/p/branch?branch=update/tree-shaking-20220308) looks good with the latest changes, and Odyssey Stats works as expected on my local test site.

🚢🚢🚢

**Pull request merged**